extends MarginContainer

@onready var _retry_button: Button = $MarginContainer/VBoxContainer/RetryButton
@onready var _main_menu_button: Button = $MarginContainer/VBoxContainer/MainMenuButton
@onready var _resource_text: RichTextLabel = $MarginContainer/VBoxContainer/RichTextLabel2
@onready var _animation: AnimationPlayer = $AnimationPlayer


func _ready():
	_retry_button.pressed.connect(_on_retry_pressed)
	_main_menu_button.pressed.connect(_on_main_menu_pressed)
	Events.player_died.connect(_on_player_death)
	Events.power_produced.connect(_on_power_produced)

	hide()


func _on_retry_pressed() -> void:
	get_tree().paused = false
	get_tree().reload_current_scene()


func _on_main_menu_pressed() -> void:
	get_tree().paused = false
	get_tree().change_scene_to_file("res://MainMenu.tscn")


func _on_player_death() -> void:
	get_tree().paused = true
	show()

	_animation.play("death")


func _on_power_produced(total_produced: int, _produced: int, _used: int) -> void:
	_resource_text.text = "Extracted Resources: " + str(total_produced)
